src/Sylius/Bundle/CoreBundle/Console/Command/InstallDatabaseCommand.php

```diff
@@ -45,21 +58,34 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $this->getEnvironment(),
         ));
 
+        /** @var QuestionHelper $questionHelper */
+        $questionHelper = $this->getHelper('question');
+
         $commands = $this
-            ->getContainer()
-            ->get('sylius.commands_provider.database_setup')
-            ->getCommands($input, $output, $this->getHelper('question'))
+            ->databaseSetupCommandsProvider
+            ->getCommands($input, $output, $questionHelper)
         ;
 
-        $this->runCommands($commands, $output);
+        $commandExecutor = new CommandExecutor($input, $output, $this->getApplication());
+
+        $commandExecutor->runCommands($commands, $output);
```

Comment:
If you keep the AbstractInstallCommand, I am wondering if we leave the runCommands inside the CommandExecutor.